### ISSUE TITLE
Keras autoencoder performing very very bad 

### ISSUE THREAD

**Issue by SaheliDe** [keras-team/keras#10663]
I made a keras autoencoder to encode biological sequences (miRNA/mRNA) , and then connected it to a RNN , but whatever I do I am getting bad results. Can you please check the autoencoder and the code and see if there is anything I am missing.

PS- I tried changing every parameter, optimizer , learning rate , metrics , layer etc but nothing works . I think there is something going wrong in the architecture.`
```
inputs = Input(shape=(timesteps, input_dim,))
    encoded = GRU(30, return_sequences=True)(inputs)
    encoded = Dropout(0.1)(encoded)
    encoded = GRU(latent_dim)(encoded)

    decoded = RepeatVector(timesteps)(encoded)
    decoded = GRU(input_dim, return_sequences=True)(decoded)

    sequence_autoencoder = Model(inputs, decoded)
    encoder = Model(inputs, encoded)

    autoencoder = Model(inputs, decoded)
    autoencoder.compile(optimizer='adam', loss='msle', metrics=['acc'])
    return autoencoder, encoder, sequence_autoencoder

#Creating encodings#

print("creating miRNA encodings...")
    miRNA_autoencoder, miRNA_encoder, miRNA_decoder = create_lstm_autoencoder(miRNA_input_dim, miRNA_timesteps, autoencoder_latent_dim)
    if os.path.exists(miRNA_weights): miRNA_autoencoder.load_weights(miRNA_weights)
    miRNA_autoencoder.fit(miRNA_train, miRNA_train, epochs=autoencoder_epochs, batch_size=autoencoder_batch_size, verbose=1)
    miRNA_autoencoder.save_weights(miRNA_weights)

### Final RNN Layer

def create_lstm_binary(input_length, rnn_hidden_dim = RNN_HIDDEN_DIM, output_dim = OUTPUT_DIM, input_dim = INPUT_DIM, dropout = DROPOUT_RATIO):
    model = Sequential()
    model.add(Dense(input_dim = input_dim, output_dim = output_dim, input_length = input_length, name='embedding_layer'))
    #model.add(Dense(input_length, input_shape=(i)))
    #model.add(Flatten())
    model.add(GRU(rnn_hidden_dim, return_sequences=True))
    model.add(Dropout(dropout))
    model.add(GRU(rnn_hidden_dim))
    model.add(Dropout(dropout))
    model.add(Dense(1, activation='softmax'))
    model.compile('adam', 'binary_crossentropy', metrics=['accuracy', ])
    return model

```
`
